scenario/state.py

```diff
@@ -609,6 +610,73 @@ class Mount(_DCBase):
     src: Union[str, Path]
 
 
+def _now_utc():
+    return datetime.datetime.now(tz=datetime.timezone.utc)
+
+
+# Ideally, this would be a subclass of pebble.Notice, but we want to make it
+# easier to use in tests by providing sensible default values, but there's no
+# default key value, so that would need to be first, and that's not the case
+# in pebble.Notice, so it's easier to just be explicit and repetitive here.
+@dataclasses.dataclass(frozen=True)
+class PebbleNotice(_DCBase):
+    key: str
+    """The notice key, a string that differentiates notices of this type.
+
+    This is in the format ``domain/path``; for example:
+    ``canonical.com/postgresql/backup`` or ``example.com/mycharm/notice``.
+    """
+
+    notice_id: str = dataclasses.field(default_factory=lambda: str(uuid4()))
```

Comment:
This is called `id` in `pebble.Notice`. It's the only field that has a different name from that type, so we should probably change it to be consistent.

Also, in real Pebble these are auto-incrementing integers (as a string), starting with `"1"` (per container). Should we mimic that?